tests/Nox.Types.Tests/Types/DateTimeRange/DateTimeRangeTests.cs

```diff
@@ -0,0 +1,322 @@
+    {
+        var dateTime = DateTime.Parse(dateTimeStr);
+        var dateTimeRange = DateTimeRange.From(new DateTime(2023, 01, 01), new DateTime(2024, 01, 01));
+
+        Assert.True(dateTimeRange.Contains(dateTime));
+    }
+
+    [Theory]
+    [InlineData("2022-12-31")]
+    [InlineData("2024-01-02")]
+    [InlineData("2021-06-06")]
+    public void Nox_DateTimeRange_Contains_ReturnsFalse(string dateTimeStr)
+    {
+        var dateTime = DateTime.Parse(dateTimeStr);
+        var dateTimeRange = DateTimeRange.From(new DateTime(2023, 01, 01), new DateTime(2024, 01, 01));
+
+        Assert.False(dateTimeRange.Contains(dateTime));
+    }
+
+    [Theory]
+    [InlineData("2023-01-01", "2023-06-01", "2023-02-01", "2023-04-01", "2023-02-01", "2023-04-01")]
+    [InlineData("2023-01-01", "2023-04-01", "2023-02-01", "2023-06-01", "2023-02-01", "2023-04-01")]
+    [InlineData("2023-02-01", "2023-06-01", "2023-01-01", "2023-04-01", "2023-02-01", "2023-04-01")]
+    [InlineData("2023-01-01", "2023-02-01", "2023-02-01", "2023-04-01", "2023-02-01", "2023-02-01")]
+    public void Nox_DateTimeRange_Intersect_ReturnsIntersectedDateTimeRange(string startStr1, string endStr1, string startStr2, string endStr2, string expectedStartStr, string expectedEndStr)
+    {
+        var start1 = DateTime.Parse(startStr1);
+        var end1 = DateTime.Parse(endStr1);
+
+        var start2 = DateTime.Parse(startStr2);
+        var end2 = DateTime.Parse(endStr2);
+
+        var expectedStart = DateTime.Parse(expectedStartStr);
+        var expectedEnd = DateTime.Parse(expectedEndStr);
+
+        var dateTimeRange1 = DateTimeRange.From(start1, end1);
+        var dateTimeRange2 = DateTimeRange.From(start2, end2);
+
+        var intersection = dateTimeRange1.Intersect(dateTimeRange2);
+        var intersection2 = dateTimeRange2.Intersect(dateTimeRange1);
+
+        Assert.Equal(intersection, intersection2);
+        Assert.NotNull(intersection);
+        Assert.Equal(expectedStart, intersection.Start);
+        Assert.Equal(expectedEnd, intersection.End);
+    }
+
+    [Fact]
+    public void Nox_DateTimeRange_Intersect_ReturnsNull()
+    {
+        var start1 = new DateTime(2023, 01, 01);
+        var end1 = new DateTime(2023, 02, 01);
+
+        var start2 = new DateTime(2021, 03, 01);
+        var end2 = new DateTime(2021, 04, 01);
+
+        var dateTimeRange1 = DateTimeRange.From(start1, end1);
+        var dateTimeRange2 = DateTimeRange.From(start2, end2);
+
+        var intersection = dateTimeRange1.Intersect(dateTimeRange2);
+        var intersection2 = dateTimeRange2.Intersect(dateTimeRange1);
+
+        Assert.Equal(intersection, intersection2);
+        Assert.Null(intersection);
+    }
+
+    [Theory]
+    [InlineData("en-GB", "20/06/2023 10:05:00 - 20/08/2023 10:05:00")]
+    [InlineData("en-US", "6/20/2023 10:05:00 AM - 8/20/2023 10:05:00 AM")]
+    public void Nox_DateTimeRange_ToString_WithoutParameters_ReturnsString(string culture, string expectedResult)
```

Comment:
yes this is a good test, needs to be updated according the Tostring() Expectations